database/migrations/2022_04_15_130255_create_availabilities_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateAvailabilitiesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('availabilities', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('vehicle_id')->nullable();
+            $table->dateTime('start_time');
+            $table->dateTime('end_time');
+            $table->float('latitude');
+            $table->float('longitude');
+            $table->timestamps();
+
+            // Foreign key constraints (when deleting a vehicle, nullify its availabilities)
+            $table->foreign('vehicle_id')->references('vehicle_id')->on('vehicles')->onDelete('set null');
```

Comment:
Bi li možda bilo praktičnije brisati i availabilityje obrisanih vozila, budući da je ovo više "utility" tablica nego nešto čije bismo podatke trebali arhivirati? Kada bismo imali velik broj vozila u bazi s velikim brojem zapisa dostupnosti, količina zastarjelih podataka bi mogla biti poprilična.
```suggestion
            $table->foreign('vehicle_id')->references('vehicle_id')->on('vehicles')->onDelete('cascade');
```

Reply:
Prema mom iskustvu, brisanju podataka je uvijek bolje pristupiti s oprezom.
U ovom slučaju, availability nam se direktno veže s rentom koji je ključna tablica za funkcionalnost naše aplikacije. Zbog toga što će svaka instanca renta biti, ako smo se dobro shvatili, limitiranja vremenskim okvirom availabilityja mislim da bi bilo dobro zadržati informaciju koliki je taj vremenski okvir bio.

Kasnije možemo implementirati automatsko brisanje ili arhiviranje za podatke koji zaista više nemaju svrhu (npr. sva vozila i availabilityje za usera koji je potpuno obrisao account)